File: examples/java-app/src/main/java/com/example/Calculator.java
package com.example;

public class Calculator {

    public int add(int a, int b) {
        return a + b;
    }

    public double divide(int a, int b) {
        if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }
        return (double) a / b;
    }

    public boolean isEven(int n) {
        return n % 2 == 0;
    }
}

File: examples/python-app/src/calculator.py
"""Calculator module — a simple demo for GitHub Actions CI."""

def add(a: float, b: float) -> float:
    """Return the sum of two numbers."""
    return a + b


def divide(a: float, b: float) -> float:
    """Return a divided by b. Raises ValueError on division by zero."""
    if b == 0:
        raise ValueError("Division by zero is not allowed")
    return a / b


def is_even(n: int) -> bool:
    """Return True if n is even."""
    return n % 2 == 0
